packages/playground/cli/src/xdebug-path-mappings.ts

```diff
@@ -0,0 +1,364 @@
+			// If auto-mounting from the current directory,
+			// the entire project directory can be mapped.
+			absoluteHostPath === cwd ||
+			absoluteHostPath.startsWith(cwdChildPrefix)
+		);
+	});
+}
+
+export type IDEConfig = {
+	/**
+	 * The name of the configuration within the IDE configuration.
+	 */
+	name: string;
+	/**
+	 * The IDEs to configure.
+	 */
+	ides: string[];
+	/**
+	 * The web server host.
+	 */
+	host: string;
+	/**
+	 * The web server port.
+	 */
+	port: number;
+	/**
+	 * The mounts to consider for debugger path mapping.
+	 */
+	mounts: Mount[];
+};
+
+/**
+ * Implement necessary parameters and path mappings in IDE configuration files.
+ *
+ * @param name The configuration name.
+ * @param mounts The Playground CLI mount options.
+ */
+export async function addXdebugIDEConfig({
+	name,
+	host,
+	port,
+	ides,
+	mounts,
+}: IDEConfig) {
+	const mappings = filterLocalMounts(mounts);
+
+	// PHPstorm
+	if (ides.includes('phpstorm')) {
+		const serverConfig = {
+			$: {
+				name: name,
+				// TODO: Document why host:port and port: 80 are necessary for PhpStorm to hit breakpoints?
+				// IOW, why do we not set the web server port in the port field,
+				// and what is port 80 for when we aren't opening port 80 at all?
+				host: `${host}:${port}`,
+				port: '80',
+				use_path_mappings: 'true',
+			},
+			path_mappings: [
+				{
+					mapping: mappings.map((mapping) => ({
+						$: {
+							'local-root': `$PROJECT_DIR$/${mapping.hostPath.replace(
+								/^\.\/?/,
+								''
+							)}`,
+							'remote-root': mapping.vfsPath,
+						},
+					})),
+				},
+			],
+		};
+
+		const configFilePath = path.join(process.cwd(), '.idea/workspace.xml');
+
+		if (!fs.existsSync(configFilePath)) {
+			const dirname = path.dirname(configFilePath);
+			if (!fs.existsSync(dirname)) {
+				if (ides.length > 1) return;
+
+				fs.mkdirSync(dirname);
+			}
+			fs.writeFileSync(
+				configFilePath,
+				'<?xml version="1.0" encoding="UTF-8"?>\n<project version="4">\n</project>'
+			);
+		}
+
+		const contents = fs.readFileSync(configFilePath);
+		const config = await parseStringPromise(contents);
+
+		if (!config.project) {
+			config.project = { $: { version: '4' }, component: [] };
+		}
+
+		let component = config?.project?.component?.find(
+			(c: any) => c?.$?.name === 'PhpServers'
+		);
+		if (!component) {
+			component = {
+				$: { name: 'PhpServers' },
+				servers: [{ server: [] }],
+			};
+			if (!config.project.component) {
+				config.project.component = [];
+			}
+			config.project.component.push(component);
+		} else {
+			// Sometimes, existing configs can be edited to remove sections
+			// unexpectedly. Let's ensure the right child elements exist
+			// before saving our server config.
+
+			if (!component.servers) {
+				component.servers = [];
+			}
+			if (!component.servers[0]) {
+				component.servers[0] = { server: [] };
+			}
+			if (!component.servers[0].server) {
+				component.servers[0].server = [];
+			}
```

Comment:
It is really easy to run into errors if we assume too much about the existing structure. These files are editable by users, so the purpose of this code is to ensure our assumptions about XML structure will hold.

It feels like there ought to be a clearer, more succinct way to do this, but this at least works.